<launch>
	<arg name="x" default="0.0" />
	<arg name="y" default="0.0" />
	<arg name="z" default="0.0" />
	<arg name="R" default="0.0" />
	<arg name="P" default="0.0" />
	<arg name="Y" default="0.0" />
  <arg name="frame" default="base_footprint" />
  <arg name="worldFrame" default="world_frame" />
  <arg name="drone_index" default="1" />

<param name="robot_description" command="$(find xacro)/xacro '$(find espdrone_driver)/urdf/espdrone.urdf.xacro' drone_index:=$(arg drone_index)"/>

  <node
    name="spawn_model"
    pkg="gazebo_ros"
    type="spawn_model"
    respawn="false"
    args="-model drone$(arg drone_index) -urdf -param robot_description -x $(arg x) -y $(arg y) -z $(arg z) -R $(arg R) -P $(arg P) -Y $(arg Y)"
    output="screen" />

  <!--node name="controller" pkg="espdrone_controller" type="controller" output="screen" ns="/$(arg name)" >
    <param name="frame" value="/$(arg name)_tf/$(arg frame)" />
    <param name="worldFrame" value="$(arg worldFrame)" />
    <rosparam command="load" file="$(find espdrone_controller)/config/espdrone.yaml" />
  </node-->

  <node name="joint_state_publisher" type="joint_state_publisher" pkg="joint_state_publisher" />
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" ns="/drone$(arg drone_index)" args="joint1_velocity_controller joint2_velocity_controller joint3_velocity_controller joint4_velocity_controller joint_state_controller"/>
  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
    respawn="false" output="screen">
    <remap from="/joint_states" to="/drone$(arg drone_index)/joint_states" />
  </node>
  
<<<<<<< HEAD
  <!--<include file="$(find espdrone_aruco_bringup)/launch/espdrone_aruco.launch" >
    <arg name="drone_index" value="$(arg drone_index)" />
  </include> -->
=======
  <include file="$(find espdrone_aruco_bringup)/launch/espdrone_aruco.launch" >
    <arg name="drone_index" value="$(arg drone_index)" />
  </include>
>>>>>>> 594ff9c985efebc4ef2df35b4159c5afd7d44ff9

</launch>
